### DISCUSSION TITLE
컨트롤러 걷기 제어

### DISCUSSION THREAD

**Discussion by Park-junseo** [naverz/zepeto-studio-kor#1227] [4. General]
컨트롤러를 조작해서 아무리 조이스틱 세게 땡기거나 방향키를 조작해도 걷는 정도의 벡터값만 수정하여 입력되게 하고 싶습니다.
이유는 제페토 예제에 있는 셀카모드를 적용시켰을 때, 걷게 하고 싶은데, 그러려면 혹시 컨트롤러를 새로 만들어 주어야 하나요?

**Comment by zlmrzlo** [comment]
카테고리를 5번 질문으로 바꾸시면 더 많은 월드 개발자분들이 답변하기 쉬워질 것 같아요.
---

셀카모드 상태일 때 뛰는 애니메이션의 활성화 임계치를 런타임으로 높여놓고 
셀카모드가 아닌 상태일 때 다시 원래대로 돌려놓으면 될 것 같기도 하네요

**Reply by Park-junseo** [reply]
그러면 player character currentState는 그대로라서 멀티 플레이 상황에서 서버에 잘못된 상태를 보내게 되기도 하고,
또한 controller에 runSpeed를 낮추면, 다른 플레이어 인스턴스도 뛰어야 하는 상태에서 느리게 뛰게 됩니다.
그래서 컨트롤하는 클라이언트 상황에서만 자신이 컨트롤하는 인스턴스에 move Vector 값을 낮게 입력하게 하여 셀카 모드일 때는 걷게 하고 싶습니다.

**Comment by Zepeto-TechEvangelist** [comment]
안녕하세요, 셀피모드일때 Walk Threshold 값을 1로 변경하는 방법을 사용하시기를 추천 드립니다.

**Reply by Park-junseo** [reply]
[](https://github.com/naverz/zepeto-studio-kor/discussions/1227#discussioncomment-3491418)
walkThreshold는 ZepetoPlayer 클래스의 멤버로그것을 변경하면 멀티환경에서의 다른 플레이어 인스턴스도 영향을 미치게 됩니다.

**Reply by Zepeto-TechEvangelist** [reply]
로컬플레이어에게만 적용되도록 조건을 추가해 보세요.

**Reply by Park-junseo** [reply]
안녕하세요
walkThreshold는 ZepetoPlayers.instance.motionV2Data.walkThreshold = 1; 이런 식으로 바꿀 수 있을 텐데, 이러면 멀티 환경에서 다른 플레이어도 뛰어야 하는 거리를 걷는 모션이 되게 됩니다.
ZepetoPlayers.instance.motionV2Data.runSpeed = ZepetoPlayers.instance.motionV2Data.walkSpeed; 를 적용하면 마찬가지로 다른 플레이어도 뛰어야 하는 모션에서 느려지게 됩니다. 로컬플레이 적용이 따로 있나요?

**Reply by zlmrzlo** [reply]
네, 로컬플레이한테만 속성값을 적용시킬 수 있습니다.
ClientStarter에서 OnStateChange() 함수 부분을 보면 로컬 플레이와 아닌 것을 구분하는 코드가 존재합니다.
이 부분을 참고하여 개발에 이용하면 될 것 같아요.

**Reply by Park-junseo** [reply]
네 제가 알기론 onStateChange와 상관 없이 OnAddedLocalPlayer.AddListener 나 OnAddedPlayer.AddListener 를 말씀하시는 것 같은데, 물론 이것으로 로컬플레이가 아닌 것을 구분할 수 있지만, 모든 플레이어 인스턴스에 대한 walkThreshold는 변경할 수는 있어도 한 플레이어에게만 따로 walkThreshold는 지정할 수 없는 것으로 보입니다. 답변 부탁드립니다. 제가 잘 못 알고 있는 건가요..ㅜㅜ
